aea/cli/config.py

```diff
@@ -54,89 +63,269 @@ def get(ctx: Context, json_path: List[str]):
-    if not isinstance(parent_obj.get(attr_name), (str, int, bool, float)):
-        raise click.ClickException(
-            "Attribute '{}' is not of primitive type.".format(attr_name)
+
+    def _get_value_from_configuration_object(self, conf_obj: Dict) -> Any:
+        """Get value from configuration object."""
+        return self._get_parent_object(conf_obj).get(self.attr_name)
+
+    def _get_parent_object(self, conf_obj: Dict) -> Dict:
+        """
+        Get and validate parent object.
+
+        :param conf_obj: configuration object.
+
+        :return: parent object.
+        :raises: ClickException if attribute is not valid.
+        """
+        parent_obj_path = self.json_path[:-1]
+        attr_name = self.attr_name
+        try:
+            parent_obj = get_parent_object(conf_obj, parent_obj_path)
+        except ValueError as e:
+            raise click.ClickException(str(e))
+
+        if attr_name not in parent_obj:
+            raise click.ClickException("Attribute '{}' not found.".format(attr_name))
+        if not isinstance(parent_obj.get(attr_name), (str, int, bool, float)):
+            raise click.ClickException(
+                "Attribute '{}' is not of primitive type.".format(attr_name)
+            )
+        return parent_obj
+
+    @property
+    def agent_config(self) -> AgentConfig:
+        """Return current context AgentConfig."""
+        return self.ctx.agent_config
+
+    def _check_set_field_name(self) -> None:
+        """
+        Check field names on set operation.
+
+        :return: None
+
+        :raises: click.ClickException is field is not allowed to be changeed.
+        """
+        if (
+            self.is_target_agent
+            and self.json_path[0] in self.AGENT_FIELDS_ALLOWED_TO_CHANGE
+        ):
+            return
+        if self.json_path[0] in self.FIELDS_NOT_ALLOWED_TO_CHANGE:
+            raise click.ClickException(
+                f"Field `{self.json_path[0]}` is not allowed to change!"
+            )
+
+    def _fix_component_id_version(self) -> None:
+        """Update self.component_id with actual version defined in agent instead of latest."""
+        if not self.component_id:  # pragma: nocover: check for mypy
+            raise ValueError("Component id is not set")
+
+        component_id = None
+
+        for component_id in self.agent_config.package_dependencies:
+            if (
+                component_id.author == self.component_id.author
+                and component_id.package_type == self.component_id.package_type
+                and component_id.name == self.component_id.name
+            ):
+                break
+        else:  # pragma: nocover  # should be always ok, cause component has to be alrady registered
+            raise ValueError("component is not registered?")
+
+        self.component_id = component_id
+
+    def _parent_object_for_agent_component_configuration(self) -> Dict:
```

Comment:
Missing docstring, incomplete typing

Reply:
i can not type dict object, cause it depends  configuration and varibale